src/gap.c

```diff
@@ -643,24 +643,18 @@ static Obj FuncSizeScreen(Obj self, Obj args)
   }
   else {
     elm = ELMW_LIST(size,1);
-    if (!IS_INTOBJ(elm)) {
-        ErrorMayQuit("SizeScreen: <x> must be an integer", 0, 0);
-    }
-    len = INT_INTOBJ( elm );
+    len = GetSmallIntEx("SizeScreen", elm, "<x>");
     if ( len < 20  )  len = 20;
     if ( MAXLENOUTPUTLINE < len )  len = MAXLENOUTPUTLINE;
   }
 
   /* extract the number                                                  */
-  if ( LEN_LIST(size) < 2 || ELM0_LIST(size,2) == 0 ) {
+  elm = ELM0_LIST(size, 2);
```

Comment:
Note that ELM0_LIST is explicitly callable with arbitrary positive indices; if the index is out of bounds, it returns 0. This allows us to simplify this code, and avoid the call to `ELMW_LIST` later on.